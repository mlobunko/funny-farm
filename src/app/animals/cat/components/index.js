import Cat from './CatContainer';

export default Cat;
