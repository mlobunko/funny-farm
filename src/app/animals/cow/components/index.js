import Cow from './CowContainer';

export default Cow;
