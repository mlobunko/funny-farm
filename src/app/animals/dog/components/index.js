import Dog from './DogContainer';

export default Dog;
