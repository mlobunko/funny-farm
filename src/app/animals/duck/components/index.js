import Duck from './DuckContainer';

export default Duck;
