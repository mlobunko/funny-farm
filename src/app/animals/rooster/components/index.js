import Rooster from './RoosterContainer';

export default Rooster;
