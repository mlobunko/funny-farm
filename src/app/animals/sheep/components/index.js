import Sheep from './SheepContainer';

export default Sheep;
