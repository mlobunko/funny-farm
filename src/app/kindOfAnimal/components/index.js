import KindOfAnimal from './KindOfAnimalContainer';

export default KindOfAnimal;
