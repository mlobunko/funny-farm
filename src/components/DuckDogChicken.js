import React from "react";
import { connect } from "react-redux";
import { playChickenSound, playDogSound, playDuckSound } from "../actions/saga";

export const DuckDogChicken = ({
  playChickenSound,
  playDogSound,
  playDuckSound,
  isChickenSoundPlaying,
  isDogSoundPlaying,
  isDuckSoundPlaying
}) => (
  <div className="duck-dog-chicken">
    <div
      className={isDuckSoundPlaying ? "duck--chat" : "duck"}
      onClick={playDuckSound}
    />
    <div
      className={isDogSoundPlaying ? "dog--chat" : "dog"}
      onClick={playDogSound}
    />
    <div
      className={isChickenSoundPlaying ? "chicken--chat" : "chicken"}
      onClick={playChickenSound}
    />
  </div>
);

const mapStateToProps = state => ({
  isChickenSoundPlaying: state.settings.isChickenSoundPlaying,
  isDogSoundPlaying: state.settings.isDogSoundPlaying,
  isDuckSoundPlaying: state.settings.isDuckSoundPlaying
});

export default connect(mapStateToProps, {
  playChickenSound,
  playDogSound,
  playDuckSound
})(DuckDogChicken);
