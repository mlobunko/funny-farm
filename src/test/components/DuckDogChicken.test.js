import React from "react";
import { shallow } from "enzyme";
import { DuckDogChicken } from "../../components/DuckDogChicken";
import {
  playChickenSound,
  playDogSound,
  playDuckSound
} from "../../actions/saga";

it("should render DuckSheepChicken component correctly", () => {
  const wrapper = shallow(
    <DuckDogChicken
      playChickenSound={playChickenSound}
      playDogSound={playDogSound}
      playDuckSound={playDuckSound}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it("should call playChickenSound on button click", () => {
  const playChickenSound = jest.fn();
  const wrapper = shallow(
    <DuckDogChicken
      playChickenSound={playChickenSound}
      playDogSound={playDogSound}
      playDuckSound={playDuckSound}
    />
  );
  wrapper.find(".chicken").simulate("click");
  expect(playChickenSound).toHaveBeenCalledTimes(1);
});

it("should call playDuckSound on button click", () => {
  const playDuckSound = jest.fn();
  const wrapper = shallow(
    <DuckDogChicken
      playChickenSound={playChickenSound}
      playDogSound={playDogSound}
      playDuckSound={playDuckSound}
    />
  );
  wrapper.find(".duck").simulate("click");
  expect(playDuckSound).toHaveBeenCalledTimes(1);
});

it("should call playDogSound on button click", () => {
  const playDogSound = jest.fn();
  const wrapper = shallow(
    <DuckDogChicken
      playChickenSound={playChickenSound}
      playDuckSound={playDuckSound}
      playDogSound={playDogSound}
    />
  );
  wrapper.find(".dog").simulate("click");
  expect(playDogSound).toHaveBeenCalledTimes(1);
});
