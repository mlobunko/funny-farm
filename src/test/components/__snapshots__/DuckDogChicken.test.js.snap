// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DuckSheepChicken component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    playChickenSound={[Function]}
    playDogSound={[Function]}
    playDuckSound={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          aria-label="play duck sound button"
          className="duck"
          onClick={[Function]}
/>,
        <div
          aria-label="play dog sound button"
          className="dog"
          onClick={[Function]}
/>,
        <div
          aria-label="play rooster sound button"
          className="chicken"
          onClick={[Function]}
/>,
      ],
      "className": "duck-dog-chicken",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "play duck sound button",
          "className": "duck",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "play dog sound button",
          "className": "dog",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "play rooster sound button",
          "className": "chicken",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            aria-label="play duck sound button"
            className="duck"
            onClick={[Function]}
/>,
          <div
            aria-label="play dog sound button"
            className="dog"
            onClick={[Function]}
/>,
          <div
            aria-label="play rooster sound button"
            className="chicken"
            onClick={[Function]}
/>,
        ],
        "className": "duck-dog-chicken",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "play duck sound button",
            "className": "duck",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "play dog sound button",
            "className": "dog",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "play rooster sound button",
            "className": "chicken",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
